src/components/Image/index.js

```diff
@@ -7,76 +7,66 @@ import ONYXKEYS from '../../ONYXKEYS';
 import {defaultProps, imagePropTypes} from './imagePropTypes';
 import RESIZE_MODES from './resizeModes';
 
-class Image extends React.Component {
-    componentDidMount() {
-        this.configureOnLoad();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (prevProps.source === this.props.source) {
-            return;
-        }
-        this.configureOnLoad();
-    }
-
+function Image(props) {
+    const {source: propsSource, isAuthTokenRequired, onLoad, session} = props;
     /**
      * Check if the image source is a URL - if so the `encryptedAuthToken` is appended
      * to the source.
      * @returns {Object} - the configured image source
      */
-    getImageSource() {
-        const source = this.props.source;
-        let imageSource = source;
-        if (this.props.isAuthTokenRequired) {
+    const source = useMemo(() => {
+        if (isAuthTokenRequired) {
+            const innerSource = propsSource;
```

Comment:
no needs to create an additional variable - just use propsSource directly at line 24 propsSource.uri